include/pistache/async.h

```diff
@@ -241,9 +254,14 @@ namespace Async {
 
             bool isVoid() const override { return false; }
 
+        protected:
             void *memory() override {
                 return &storage;
             }
+
+        private:
+            typedef typename std::aligned_storage<sizeof(T), alignof(T)>::type Storage;
```

Comment:
Maybe it would be better to write:
```
using Storage = typename std::aligned_storage<sizeof(T), alignof(T)>::type;
```

Reply:
I just moved the code around:
 - I want it 'private' 
- This is to make sure nobody does any funny business (that would result in dangling pointers)

If you want something prettier that can be done as a separate commit.